vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
-        for page in self.advisories[ecosystem]:
-            for adv in page["data"]["securityVulnerabilities"]["edges"]:
-                packages.add(adv["node"]["package"]["name"])
-        return packages
-
-    def process_response(self) -> List[Advisory]:
-        adv_list = []
-        for ecosystem in self.advisories:
-            self.set_version_api(ecosystem)
-            pkg_type = self.version_api.package_type
-            for resp_page in self.advisories[ecosystem]:
-                for adv in resp_page["data"]["securityVulnerabilities"]["edges"]:
-                    name = adv["node"]["package"]["name"]
-                    cutoff_time = dateparser.parse(adv["node"]["advisory"]["publishedAt"])
-                    affected_purls = []
-                    unaffected_purls = []
-                    if self.process_name(ecosystem, name):
-                        ns, pkg_name = self.process_name(ecosystem, name)
-                        if hasattr(self.version_api, "module_name_by_package_name"):
-                            pkg_name = self.version_api.module_name_by_package_name.get(
-                                name, pkg_name
-                            )
-                        aff_range = adv["node"]["vulnerableVersionRange"]
-                        aff_vers, unaff_vers = self.categorize_versions(
-                            self.version_api.package_type,
-                            aff_range,
-                            self.version_api.get(name, until=cutoff_time).valid_versions,
-                        )
-                        affected_purls = [
-                            PackageURL(name=pkg_name, namespace=ns, version=version, type=pkg_type)
-                            for version in aff_vers
-                        ]
+                version_range_class: VersionRange = ECOSYSTEM_VERSION_RANGE_MAPPING.get(ecosystem)
+                affected_range = version_range_class.from_native(aff_range)
+            except InvalidVersionRange:
+                LOGGER.error(f"Could not parse affected range {aff_range!r}")
+                continue
+            affected_packages.append(
+                AffectedPackage(
+                    package=PackageURL(
+                        name=pkg_name, namespace=ns, type=get_type_from_ecosystem(ecosystem)
+                    ),
+                    affected_version_range=affected_range,
+                )
+            )
+        advisory = get_item(node, "advisory")
+        if not advisory:
+            LOGGER.error(f"No advisory found in {node!r}")
+            continue
+        references = get_item(advisory, "references") or []
+        if references:
+            references: List[Reference] = list(extract_references(references))
+        vuln_desc = get_item(advisory, "summary")
+        identifiers = get_item(advisory, "identifiers") or []
+        for identifier in identifiers:
+            # collect CVEs
+            if identifier["type"] == "CVEjson_object":
```

Comment:
I think you fat fingered this:
```suggestion
            if identifier["type"] == "CVE":
```